File: PA2/02/fibonacci_last_digit.py
# Uses python3
import sys

def get_fibonacci_last_digit_naive(n):
    if n <= 1:
        return n

    previous = 0
    current  = 1

    for _ in range(n - 1):
        # JDH original previous, current = current, previous + current
        previous, current = (current % 10), (previous + current) % 10   # JDH added

    return current # JDH was 5 10

if __name__ == '__main__':
    input = sys.stdin.readline()    # JDH was .read()
    n = int(input)
    print(get_fibonacci_last_digit_naive(n))

File: PA2/02/fibonacci_last_digit.cpp
#include <iostream>

int get_fibonacci_last_digit_naive(int n) {
    if (n <= 1)
        return n;

    int previous = 0;
    int current  = 1;

    for (int i = 0; i < n - 1; ++i) {
        int tmp_previous = previous % 10;	// JDH added % 10
        previous = current;
        current = (tmp_previous + current)%10;	// JDH added %10
    }

    return current;					// JDH was % 10
}

int main() {
    int n;
    std::cin >> n;
    int c = get_fibonacci_last_digit_naive(n);
    std::cout << c << '\n';
    }